ui-v2/src/components/concurrency/global-concurrency-view/index.tsx

```diff
@@ -1,28 +1,58 @@
 import { useListGlobalConcurrencyLimits } from "@/hooks/global-concurrency-limits";
-import { useState } from "react";
-
+import { useMemo, useState } from "react";
+import { CreateOrEditLimitDialog } from "./create-or-edit-limit-dialog";
+import { GlobalConcurrencyLimitEmptyState } from "./global-concurrency-limit-empty-state";
 import { GlobalConcurrencyLimitsHeader } from "./global-concurrency-limits-header";
 
+type AddOrEditDialogState = {
+	open: boolean;
+	limitIdToEdit?: string;
+};
+
 export const GlobalConcurrencyView = () => {
-	const [showAddDialog, setShowAddDialog] = useState(false);
+	const [openAddOrEditDialog, setOpenAddOrEditDialog] =
+		useState<AddOrEditDialogState>({
+			open: false,
+		});
 
 	const { data } = useListGlobalConcurrencyLimits();
 
-	const openAddDialog = () => setShowAddDialog(true);
-	const closeAddDialog = () => setShowAddDialog(false);
+	const selectedlimitToUpdate = useMemo(() => {
+		if (!openAddOrEditDialog.limitIdToEdit) {
+			return undefined;
+		}
+		return data.find((limit) => limit.id === openAddOrEditDialog.limitIdToEdit);
+	}, [data, openAddOrEditDialog.limitIdToEdit]);
```

Comment:
We might want to revisit this once the table is created. We should be able to have the table put the full data in a callback, which would negate the need to have a `.find` call.

Reply:
I'll keep an eye out for it, because this approach is to help eliminate duplicate state - which can lead to UX bugs.

eg problem case for duplicating state:
1. UI gets list of data from `useSuspenseQuery`.
2. UI duplicates selected data into a `useState`.
3. `useSuspenseQuery` refetches, and updates its state of that selected data. (from some async event). The UI is now working with stale data.

When using this memoize approach.
1. UI gets list of data from `useSuspenseQuery`.
2. UI keeps track of the selected data by id.
3. Selected data is found by `.find()`.
4. `useSuspenseQuery` refetches, and updates its state of that selected data. (from some async event). 
5. the selected data that calls `.find()` is ran because the `useMemo` dependency has changed from the async event. The UI is now working with the updated data source

Reply:
That makes sense. I don't think we'll need to worry about stale data from the table because we can directly pass in the data from our query to `useTable` and [retrieve the original data for the row](https://tanstack.com/table/latest/docs/guide/rows#access-original-row-data) to avoid working with duplicated data.

Reply:
oh nice, that makes sense

Reply:
I'll update that when I implement the table next